tests/comments/test_async_serializer.py

```diff
@@ -0,0 +1,229 @@
+
+    # Testing a plain comment
+    comment_1 = comment_data[6]
+    assert comment_1['id'] == 1
+    assert comment_1['user_name'] == ''  # should be user.username ?!!
+    # In the react_comments_async template tags,
+    # this is set as user_name = str(user.id)
+    # And in the react_comments template tags,
+    # this is set to user.username AFTER the serializer
+    assert comment_1['user_pk'] == str(user.pk)
+    # django auth user model does not define get_absolute_url()
+    assert comment_1['user_profile_url'] == ''
+    assert comment_1['is_deleted'] is False
+    assert comment_1['is_moderator'] is False
+    assert len(comment_1['child_comments']) == 0
+    assert comment_1['comment'] == Comment.objects.get(pk=1).comment
+    assert comment_1['object_pk'] == question.pk
+    assert comment_1['is_removed'] is False
+    assert comment_1['is_censored'] is False
+    assert comment_1['is_blocked'] is False
+    assert len(comment_1['comment_categories']) == 0
+    assert comment_1['last_discussed'] is None
+    assert comment_1['is_moderator_marked'] is False
+    assert comment_1['content_type'] == question_ct.pk
+
+    # Testing comment with a defined category:
+    comment_2 = comment_data[5]
+    assert comment_2['id'] == 2
+    assert comment_2['user_name'] == ''
+    assert comment_2['user_pk'] == str(user.pk)
+    assert comment_2['user_profile_url'] == ''
+    assert comment_2['is_deleted'] is False
+    assert comment_2['is_moderator'] is False
+    assert len(comment_2['child_comments']) == 0
+    assert comment_2['comment'] == Comment.objects.get(pk=2).comment
+    assert comment_2['object_pk'] == question.pk
+    assert comment_2['is_removed'] is False
+    assert comment_2['is_censored'] is False
+    assert comment_2['is_blocked'] is False
+    assert len(comment_2['comment_categories']) == 1
+    assert comment_2['comment_categories'] == {'QUE': 'Question'}
+    assert comment_2['last_discussed'] is None
+    assert comment_2['is_moderator_marked'] is False
+    assert comment_2['content_type'] == question_ct.pk
+
+    # Testing a comment that is set to blocked by a moderator:
+    comment_3 = comment_data[4]
+    assert comment_3['id'] == 3
+    assert comment_3['user_name'] == ''
+    assert comment_3['user_pk'] == str(user.pk)
+    assert comment_3['user_profile_url'] == ''
+    assert comment_3['is_deleted'] is False
+    assert comment_3['is_moderator'] is False
+    assert len(comment_3['child_comments']) == 0
+    assert comment_3['comment'] != Comment.objects.get(pk=3).comment
+    assert comment_3['comment'] == ''
+    assert comment_3['object_pk'] == question.pk
+    assert comment_3['is_removed'] is False
+    assert comment_3['is_censored'] is False
+    assert comment_3['is_blocked'] is True
+    assert len(comment_3['comment_categories']) == 0
+    assert comment_3['last_discussed'] is None
+    assert comment_3['is_moderator_marked'] is False
+    # Shouldn't a blocked comment be moderator marked?
```

Comment:
No, the moderator mark is the opposite. It is used for comments that the moderator wants to everyone to see and is used to put the comment at the top at the list with the ordering filter in one project.